pkg/backup/backupschedule/backup_schedule_manager.go

```diff
@@ -253,21 +253,16 @@ func createBackup(bkController controller.BackupControlInterface, bs *v1alpha1.B
 }
 
 func (bm *backupScheduleManager) backupGC(bs *v1alpha1.BackupSchedule) {
-	ns := bs.GetNamespace()
-	bsName := bs.GetName()
-
 	// if MaxBackups and MaxReservedTime are set at the same time, MaxReservedTime is preferred.
 	if bs.Spec.MaxReservedTime != nil {
 		bm.backupGCByMaxReservedTime(bs)
 		return
 	}
 
-	if bs.Spec.MaxBackups != nil && *bs.Spec.MaxBackups > 0 {
+	if (bs.Spec.MaxBackups != nil && *bs.Spec.MaxBackups > 0) || bs.Spec.MaxCompletedBackups != nil || bs.Spec.MaxFailedBackups != nil {
```

Comment:
Check the `*bs.Spec.MaxCompletedBackups > 0` and `*bs.Spec.MaxFailedBackups > 0`